File: 215-kth-largest-element-in-an-array/215-kth-largest-element-in-an-array.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> maxHeap = new PriorityQueue(Collections.reverseOrder());
        for(int i = 0; i < nums.length; i++){
            maxHeap.add(nums[i]);
        }
        int toPop = k-1;
        for(int i =0; i < toPop; i++){
            maxHeap.poll();
        }
        return maxHeap.peek();
    }
}

File: 215-kth-largest-element-in-an-array/215-kth-largest-element-in-an-array.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        k = nums.size()-k;
        return quickSelect(nums,0,nums.size()-1,k);
    }
    int quickSelect(vector<int>& nums, int l, int r, int k){
        int index = partition(nums,l,r);
        if(index == k){
            return nums[index];
        }
        if(index < k){
            return quickSelect(nums,index+1,r,k);
        }
        else{
            return quickSelect(nums,l,index-1,k);
        }
    }
    int partition(vector<int>& nums, int l, int r){
        int x = nums[r], i = l;
        for(int j = l; j <=r-1; j++){
            if(nums[j] <= x){
                swap(nums[i],nums[j]);
                i++;
            }
        }
        swap(nums[i],nums[r]);
        return i;
    }
};